packages/create-gasket-app/lib/scaffold/actions/load-template.js

```diff
@@ -0,0 +1,118 @@
+  const [
+    name,
+    version = 'latest'
+  ] = template.split(reNameVersion).filter(Boolean);
+
+  return {
+    name,
+    version
+  };
+}
+
+/**
+ * Safely cleans up temporary directory
+ * @param {string} tmpDir - Temporary directory path
+ * @returns {Promise<void>}
+ */
+async function cleanupTempDir(tmpDir) {
+  if (!tmpDir) return;
+
+  try {
+    await rm(tmpDir, { recursive: true });
+  } catch {
+    // Ignore cleanup errors
+  }
+}
+
+/**
+ * Downloads and installs a remote template package
+ * @param {string} template - Template name with optional version
+ * @param {string} appName - App name for temp directory
+ * @returns {Promise<{ templateDir: string, templateName: string, tmpDir: string }>} Template installation details
+ */
+async function installRemoteTemplate(template, appName) {
+  const tmpDir = await mkdtemp(path.join(os.tmpdir(), `gasket-template-${appName}`));
+  const modPath = path.join(tmpDir, 'node_modules');
+  const pkgManager = new PackageManager({
+    packageManager: 'npm',
+    dest: tmpDir
+  });
+
+  const { name, version } = parseTemplateNameAndVersion(template);
+
+  try {
+    await pkgManager.exec('install', [`${name}@${version}`]);
+    return {
+      templateDir: path.join(modPath, name, 'template'),
+      templateName: `${name}@${version}`,
+      tmpDir
+    };
+  } catch (err) {
+    await cleanupTempDir(tmpDir);
+
+    const errorMessage = err.stderr || err.message;
+    if (err.stderr?.includes('is not in this registry')) {
+      const message = `Template not found in registry: ${name}@${version}. ` +
+        'Use npm_config_registry=<registry> to use privately scoped templates.';
+      throw new Error(message);
+    }
+    throw new Error(`Failed to install template ${name}@${version}: ${errorMessage}`);
+  }
+}
+
+/**
+ * Gets template directory and name based on context
+ * @param {object} context - Gasket creation context
+ * @returns {Promise<{ templateDir: string, templateName: string, tmpDir?: string }>} Template details
+ */
+async function getTemplateDetails(context) {
+  if (context.templatePath) {
+    return {
+      templateDir: path.resolve(context.templatePath, 'template'),
+      templateName: `local template at ${path.resolve(context.templatePath, 'template')}`
+    };
+  }
+
+  const result = await installRemoteTemplate(context.template, context.appName);
+  return {
+    templateDir: result.templateDir,
+    templateName: result.templateName,
+    tmpDir: result.tmpDir
+  };
+}
+
+/** @type {import('../../internal.d.ts').loadTemplate} */
+async function loadTemplate({ context }) {
+  if (!context.template && !context.templatePath) return;
+
+  const { templateDir, templateName, tmpDir } = await getTemplateDetails(context);
+
+  // Store template information in context for other actions to use
+  context.templateDir = templateDir;
+  context.templateName = templateName;
+  context.tmpDir = tmpDir;
+
+  context.messages.push(`Template ${templateName} loaded`);
```

Comment:
I don't think this message is useful. The user would already know it's been loaded. Reserve the `context.messages` for things the user should be aware of next steps or issues duing the process.